extensions/amp-trinity-tts-player/0.1/amp-trinity-tts-player.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Layout} from '../../../src/layout';
+import {getData} from '../../../src/event-helper';
+import {isExperimentOn} from '../../../src/experiments';
+import {userAssert} from '../../../src/log';
+
+const TAG = 'amp-trinity-tts-player';
+const URL = 'https://trinitymedia.ai';
+const CDN_URL = 'https://vd.trinitymedia.ai';
+
+export class AmpTrinityTTSPlayer extends AMP.BaseElement {
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {string} */
+    this.campaignId_ = '';
+
+    this.isExperimentOn_ = isExperimentOn(this.win, 'amp-trinity-tts-player');
+  }
+
+  /** @override */
+  buildCallback() {
+    userAssert(this.isExperimentOn_, `Experiment ${TAG} is not turned on.`);
+
+    this.campaignId_ = userAssert(
+      this.element.getAttribute('data-campaign-id'),
+      'data-campaign-id attribute must be specified for <%s>',
+      TAG
+    );
+  }
+
+  /** @override */
+  layoutCallback() {
+    userAssert(this.isExperimentOn_, `Experiment ${TAG} is not turned on.`);
+
+    return new Promise(resolve => {
+      this.element.setAttribute('layout', 'fixed-height');
+
+      const iframe = this.win.document.createElement('iframe');
+      iframe.setAttribute('hidden', true);
+
+      window.addEventListener('message', event => {
+        if (getData(event)['type'] === 'trinity-player-rendered') {
+          resolve();
+          iframe.removeAttribute('hidden');
+        }
+      });
+
+      iframe.addEventListener('load', () => {
+        iframe.contentWindow./*OK*/ postMessage(
+          /** @type {JsonObject} */ ({
+            type: 'init',
+            data: {
+              text: this.win.document.body./*OK*/ innerHTML,
+              campaignId: this.campaignId_,
```

Comment:
This is all the messaging the component does.

It comes from one parameter, defined once and never changing.

Why can't this be a URL parameter on an `amp-iframe`? Could you potentially host a page that would allow your users to embed this on any iframe like `/player/trinity-amp?campaignId=FOO`?

You could also use `amp-video-iframe` to set it as a data param for author ergonomics:

```html
<amp-video-iframe 
  src="https://trinitymedia.ai/player/trinity-amp"
  data-param-campaign-id="FOO"
></amp-video-iframe>
```

It would accomplish the same.

Reply:
or you talking specifically to remove passing `campaignId`?

Reply:
Ah, just noticed the `innerHTML` reference.

I'm not sure passing all of the page's content to a third-party without user-consent follows our privacy principles.

I've copied `wg-security-privacy` on [the I2I](https://github.com/ampproject/amphtml/issues/26836) for approval. I will not make any more reviews to this PR until that's approved. Please let's continue discussion there :)

Reply:
OK, thanks @alanorozco ! Let's wait for their approval or suggesting how we should handle that in other way.